weather_mv/loader_pipeline/bq.py

```diff
@@ -162,18 +166,19 @@ def validate_arguments(cls, known_args: argparse.Namespace, pipeline_args: t.Lis
 
     def __post_init__(self):
         """Initializes Sink by creating a BigQuery table based on user input."""
-        # Define table from user input
-        if self.variables and not self.infer_schema:
-            logger.info('Creating schema from input variables.')
-            table_schema = to_table_schema(
-                [('latitude', 'FLOAT64'), ('longitude', 'FLOAT64'), ('time', 'TIMESTAMP')] +
-                [(var, 'FLOAT64') for var in self.variables]
-            )
-        else:
-            logger.info('Inferring schema from data.')
-            with open_dataset(self.example_uri, self.xarray_open_dataset_kwargs, True,
-                              self.tif_metadata_for_datetime) as open_ds:
-                ds: xr.Dataset = _only_target_vars(open_ds, self.variables)
+        with open_dataset(self.example_uri, self.xarray_open_dataset_kwargs, True,
+                          self.disable_grib_schema_normalization, self.tif_metadata_for_datetime) as open_ds_obj:
+            open_ds, is_merged_dataset = open_ds_obj
```

Comment:
Optional (recommended): consider adding `is_merged_dataset` to the `attrs` of the xarray dataset. Then, you'd only need to return one thing.

Reply:
In fact, we can always set that field whenever we open a dataset. WDYT?

Reply:
Yes, definitely this sounds great. Necessary code changes done.